errbot/backends/slack.py

```diff
@@ -650,16 +651,16 @@ def build_identifier(self, txtrep):
         log.debug("building an identifier from %s" % txtrep)
         username, userid, channelname, channelid = self.extract_identifiers_from_string(txtrep)
 
+        if userid is None and username is not None:
+            userid = self.username_to_userid(username)
+        if channelid is None and channelname is not None:
+            channelid = self.channelname_to_channelid(channelname)
+        if userid is not None and channelid is not None:
+            return SlackRoomOccupant(self.sc, userid, channelid, bot=self)
```

Comment:
I think it is wrong: a Slack Person *can* have a channel (one to one to talk to the bot).
channelids don't mean anything (if we don't test them any internal naming convention or something) as they are used for one to one and one to many discussions.

The original code uses **channelname** which actually mean you have a named slack channel to talk to the person so it is a SlackRoomOccupant.